subworkflows/local/essential_jobs/main.nf

```diff
@@ -67,19 +64,37 @@ workflow ESSENTIAL_JOBS {
 
 
     //
-    // MODULE: GENERATE A REPORT ON LENGTHS OF N's IN THE INPUT GENOME
+    // SUBWORKFLOW: GENERATE A REPORT ON LENGTHS OF N's IN THE INPUT GENOME
     //
-    TRAILINGNS (
+    TRAILINGNS_CHECK (
         FILTER_FASTA.out.fasta
     )
-    ch_versions             = ch_versions.mix(TRAILINGNS.out.versions)
+    ch_versions             = ch_versions.mix(TRAILINGNS_CHECK.out.versions)
 
 
     emit:
-    reference_tuple_from_GG = GENERATE_GENOME.out.reference_tuple
-    reference_with_seqkit   = new_input_fasta
-    trailingns_report       = TRAILINGNS.out.trailing_ns_report
-    dot_genome              = GENERATE_GENOME.out.dot_genome
-    gc_content_txt          = GC_CONTENT.out.txt
-    versions                = ch_versions
+    reference_tuple_from_GG             = GENERATE_GENOME.out.reference_tuple
+                                            .map{ meta, _file ->
+                                                def new_meta = meta + [process: "REFERENCE"]
+                                                [new_meta, _file]
+                                            }
+    reference_with_seqkit               = new_input_fasta
+    dot_genome                          = GENERATE_GENOME.out.dot_genome
+                                            .map{ meta, _file ->
+                                                def new_meta = meta + [process: "GENOME"]
+                                                [new_meta, _file]
+                                            }
+    gc_content_txt                      = GC_CONTENT.out.txt
+    trailing_ns_report                  = TRAILINGNS_CHECK.out.trailing_ns_report
+    filter_fasta_sanitation_log         = FILTER_FASTA.out.sanitation_log
+                                            .map{ meta, _file ->
+                                                def new_meta = meta + [process: "REFERENCE_SANI_LOG"]
+                                                [new_meta, _file]
+                                            }
+    filter_fasta_length_filtering_log   = FILTER_FASTA.out.length_filtering_log
+                                            .map{ meta, _file ->
+                                                def new_meta = meta + [process: "REFERENCE_FILT_LOG"]
+                                                [new_meta, _file]
+                                            }
+    versions                            = ch_versions
```

Comment:
This is a bit hard to read - can you create these as channels in the main: section and then just emit the channels?